frontend/simple/main.js

```diff
@@ -98,6 +99,26 @@ var router = new Router({
         title: 'Event Log Test Page'
       }
     },
+    /* Guards need to be created for any route that should not be directly accessed by url */
```

Comment:
This has the potential to violate DRY. See if you can create a simplified, generalized version of this to minimize *as much as possible* the potential for code duplication. I had to do the same thing in `shared/events.js` with the transformations (e.g. `ArrayToMap` etc). The result was *much much* better, more readable, better-organized, and shorter code.